x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportGetCheckpointNodeAction.java

```diff
@@ -43,12 +44,21 @@ public TransportGetCheckpointNodeAction(
 
     @Override
     protected void doExecute(Task task, Request request, ActionListener<Response> listener) {
-        getGlobalCheckpoints(indicesService, request.getShards(), listener);
+        getGlobalCheckpoints(indicesService, task, request.getShards(), listener);
     }
 
-    protected static void getGlobalCheckpoints(IndicesService indicesService, Set<ShardId> shards, ActionListener<Response> listener) {
+    protected static void getGlobalCheckpoints(
+        IndicesService indicesService,
+        Task task,
+        Set<ShardId> shards,
+        ActionListener<Response> listener
+    ) {
         Map<String, long[]> checkpointsByIndexOfThisNode = new HashMap<>();
         for (ShardId shardId : shards) {
+            if (task instanceof CancellableTask) {
+                // There is no point continuing this work if the task has been cancelled.
+                ((CancellableTask) task).ensureNotCancelled();
```

Comment:
Should this call `notifyIfCancelled` if cancelled instead of `ensureNotCancelled` given that this method returns its output through a listener instead of through a return value?

Same in `TransportGetCheckpointAction`.